### PR TITLE
Add stepstoreproduce training task

### PR DESCRIPTION
**Pull request by ayush-1506**
Fixes : #427

### CODE REVIEW

File: infra/dockerfile.train_stepstoreproduce [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM mozilla/bugbug-base:latest
+
+CMD bugbug-train stepstoreproduce
```
Comment (marco-c) [inline infra/dockerfile.train_stepstoreproduce]: Can you try if this command works for you locally? You'll probably need to update some other file in order for it to run successfully.

Comment (ayush-1506) [reply infra/dockerfile.train_stepstoreproduce]: How do I run these tasks locally?

Comment (marco-c) [reply infra/dockerfile.train_stepstoreproduce]: For the purpose of this PR, you can just run this command locally. bugbug-train is defined in setup.py, and it corresponds to scripts.trainer:main.
So you can run `python3 -c "from scripts import trainer; trainer.main()" stepstoreproduce`.

Comment (ayush-1506) [reply infra/dockerfile.train_stepstoreproduce]: Yes, this needed to be added to the trainer.py script too. I've added this now.

Comment (marco-c) [reply infra/dockerfile.train_stepstoreproduce]: Thanks. We'll need to wait for an image on Docker Hub to be created for this.

**Comment by marco-c** [comment]
@ayush-1506 could you resolve the conflicts in the PR?

File: infra/dockerfile.train_stepstoreproduce (hunk 1 above)
Comment (marco-c) [reply infra/dockerfile.train_stepstoreproduce]: The image has been added, so we can now merge this.

**Pull request merged**